src/compat.rs

```diff
@@ -4,11 +4,31 @@ use crate::{uapi, Access, AccessError, BitFlags, CompatError};
 use crate::{make_bitflags, AccessFs};
 
 /// Version of the Landlock [ABI](https://en.wikipedia.org/wiki/Application_binary_interface).
+///
+/// This `enum` may be convenient to get the features supported by a specific Linux kernel,
```

Comment:
Maybe this comment would benefit by putting the point first and the explanation for it later? For example:

> You can retrieve the supported AccessFs flags for a Landlock ABI version using AccessFs::from_all(ABI::V1).
>
> DO NOT use AccessFs::ALL or BitFlags::<AccessFs>::all() to construct a Landlock rule object -- the full
> set of AccessFs flags is a moving target and would change the semantics of your Landlock rule when
> migrating to a newer version of the Landlock crate.
>
> [longer explanation here]

?

Reply:
Updated code:
```
/// `ABI` enables to get the features supported by a specific Landlock ABI.
/// For example, [`AccessFs::from_all(ABI::V1)`](Access::from_all)
/// gets all the file system access rights defined by the first version.
///
/// Without `ABI`, it would be hazardous to rely on the the full set of access flags
/// (e.g., `BitFlags::<AccessFs>::all()` or `BitFlags::ALL`),
/// a moving target that would change the semantics of your Landlock rule
/// when migrating to a newer version of this crate
/// (i.e. non-breaking change with new supported features).
/// This usage should then be considered indeterministic because requested features
/// (e.g., access rights)
/// could not be tied to the application source code.
///
/// Such `ABI` is also convenient to get the features supported by a specific Linux kernel
/// without relying on the kernel version (which may not be accessible or patched).
```

Reply:
Nice, this is much clearer.

(A related but non-documentation question: The method call `AccessFs::from_all(ABI::V1)` reads weird to me when read left to right - is this a Rust idiosyncrasy that refers to the enum 'all' method or attribute? Why not make it more descriptive, for example `AccessFs::as_supported_in(ABI::V1)`?)

Reply:
I first used the `From` trait, which is implemented with the `from()` function to convert types. But I realized that it wasn't wise to only rely on one kind of conversion, hence the `from_all()` which is intended to express *all* defined bits from a specific ABI. In a set of WIP commits, I implemented `from_read()` and `from_write()` which split the bits in two kinds.

The *from_* part seems a bit weird now, but I would like to keep the function names short. The *new_* prefix would be misleading because `from_all()` doesn't create an `AccessFs` but a `BitFlags<AccessFs>`, which I realized is another argument to not use the *from_* prefix… The *as_* prefix has usually a specific meaning in Rust.

What about the `gen_` or `infer_` or `with_` (`AccessFs::with_all()`, `with_read`, `with_write`) prefixes? :thinking:

Reply:
Or maybe `all_from()`, `read_from()`…?

Reply:
I don't have a strong opinion here as I don't know Rust naming idioms well. I'd normally lean on calling methods in a way that the invocation reads like a sentence, but in doubt, idioms like "from()" methods trump that approach.

Reply:
Other idea: `AccessFs::all_flags(ABI::V1) -> BitFlags<AccessFs>`, `read_flags()` and `write_flags()`.

Reply:
Hmm, `read_flags()` is ambiguous because of the verb form.
`get_all_flags()`, `get_read_flags()` and `get_write_flags()` seems better.

Reply:
These `get_*()` function could be colliding with getter methods…
`new_all_flags()`, `new_read_flags()` and `new_write_flags()` seem less ambiguous (even if `new` is usually used to create a type matching the struct type, here `AccessFs`).

Reply:
I added such helpers with #10.